### PR TITLE
Implement search

### PR DESCRIPTION
**Pull request by nottrobin**
- Search box in header
- Search results page

QA
---

- Connect to VPN
- Run the site directly with Django:
  ``` bash
  ./run npm run-script watch  # Build sass
  virtualenv env
  env/bin/pip install -r requirements.txt
  env/bin/python manage.py runserver 0.0.0.0:8015
  ```

Check the header search box in multiple screen sizes, then check the search resutls page.

### CODE REVIEW

File: static/sass/_search.scss [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  @media (min-width: $breakpoint-search) {
+    border-top: 1px solid darken($color-brand, 6%);
+    background-color: transparent;
+  }
+
+  .no-js & {
+    display: block;
+  }
+
+  @media (min-width: 982px) {
```
Comment (deadlight) [inline static/sass/_search.scss:84]: Does this have a logical abstraction we could make into a variable? Something like "search threshold" or whatever its purpose is.

File: static/sass/_search.scss [hunk 2]
```diff
@@ -0,0 +1,171 @@
+.search-result {
+  border-bottom: 1px solid #cdcdcd;
+  margin-bottom: 1.5em;
+}
+
+.result-url {
+  color: #888;
```
Comment (deadlight) [inline static/sass/_search.scss:7]: Variable, pls.

File: static/sass/_search.scss [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    .no-js & {
+      display: none;
+    }
+
+    &:focus {
+      background: none;
+    }
+
+    &:hover {
+      background: rgba(0, 0, 0, 0.1);
```
Comment (deadlight) [inline static/sass/_search.scss:114]: Should be ".1"

File: static/sass/_search.scss [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+  &__back {
+    float: left;
+    padding: 0;
+    margin: 0;
+  }
+
+  &__item {
+    float: left;
+    padding: 0 0.5rem;
```
Comment (deadlight) [inline static/sass/_search.scss:26]: ".5rem"

**Review by deadlight — CHANGES_REQUESTED** [review]
Works well and looks good. There are a few code hygiene issues that'd be nice to sort out considering this is all new styling. Aside from my in-line notes, can you sort the rules alphabetically (it'll make linting less painful down the line)

**Comment by nottrobin** [comment]
Thanks @deadlight. The linter will complain about all these, and the linter need fixing as the next thing I do. I'm gonna merge this now, get it released, then fix the linter.

**Pull request merged**